ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -44,6 +43,9 @@ def check(self, _):
             self.service_check(self.SERVICE_CHECK, AgentCheck.CRITICAL, self.config.tags)
             return
 
+        if self.is_metadata_collection_enabled():
+            self.set_metadata('version', self.queue_metric_collector._collect_metadata(queue_manager))
```

Comment:
I think we shouldn't use the `semver` pattern for this case, since it's not Semver: https://semver.org/

But:
https://www.ibm.com/support/knowledgecenter/SSFKSJ_7.5.0/com.ibm.mq.ref.adm.doc/q087830_.htm

```
VVRRMMFF:
VV: Version
RR: Release
MM: Maintenance level
FF: Fix level
```

I think you can use the same metadata scheme used for `ibm_db2` (by @ChristineTChen ): https://github.com/DataDog/integrations-core/blob/8edbc8a7b5e8b0086002551dd70af18df7c177b8/ibm_db2/datadog_checks/ibm_db2/ibm_db2.py#L86-L102

It's likely that IBM use this same version pattern across all its products. From IBM integrations perspective, we should keep the same names for the parts for consistency.

```
        return {
            'major': str(int(major)),
            'minor': str(int(minor)),
            'mod': str(int(modification)),
            'fix': str(int(fix)),
        }
```